api_docs/construct-narrow.md

```diff
@@ -1,21 +1,16 @@
 # Construct a narrow
 
-A **narrow** is a set of filters for Zulip messages, that can be based
-on many different factors (like sender, stream, topic, search
-keywords, etc.). Narrows are used in various places in the the Zulip
-API (most importantly, in the API for fetching messages).
+A **narrow** is a set of filters for Zulip messages, which can be based on various factors like sender, stream, topic, search keywords, etc. Narrows are extensively used in the Zulip API, particularly in the API for fetching messages.
 
-It is simplest to explain the algorithm for encoding a search as a
-narrow using a single example. Consider the following search query
-(written as it would be entered in the Zulip web app's search box).
-It filters for messages sent to stream `announce`, not sent by
-`iago@zulip.com`, and containing the words `cool` and `sunglasses`:
+It is crucial to consider adding an `is:unread` filter to your narrow when adding the "read" flag. The Zulip server has a database index for unread messages, and including the `is:unread` filter optimizes the performance of fetching messages. This optimization is essential for any full-featured client, although it may not be obvious initially.
```

Comment:
I think this should be added further down in the text. Maybe before the **Changes** note?

And if we're adding this information to this page, then it should be revised to be clear why this is noted for the general documentation for constructing narrows. Maybe something like ...

> Clients should note that the `is:unread` filter takes advantage of the fact that there is a database index for unread messages and can be an important optimization when fetching messages in certain cases (e.g. when adding the `read` flag to a user's personal messages).

With a link to the update-message-flags-for-narrow endpoint in the example.

Reply:
I will put this before the Changes note then. Thanks for the advice, Lauryn

Reply:
The link in your updated changes is broken, see screenshot below. Also, I think the link should be in example in parenthesis since that's the direct reference to the endpoint.

![Screenshot from 2024-04-01 13-13-31](https://github.com/zulip/zulip/assets/63245456/4a666bef-e999-4253-9f11-3fffd6e63fab)

Reply:
I have fixed the link by
- putting the [] and () on the same line
- Creating a relative link instead of an absolute link
![image](https://github.com/zulip/zulip/assets/87673068/7e928301-08e5-423f-aa37-5d86ab43dfd9)